OS Provider crash when creating 'openstack_identity_application_credential_v3' resource

### Terraform Version

```
PS C:\> terraform version
Terraform v1.0.5
on windows_amd64
+ provider registry.terraform.io/terraform-provider-openstack/openstack v1.43.0
```
### Affected Resource(s)
- openstack_identity_application_credential_v3

### Terraform Configuration Files
```
resource "openstack_identity_application_credential_v3" "ac_demo_project" {
  name        = "ac_demo"
  description = "Demo project application credential"
  #  secret      = "supersecret"
  roles = ["member"]
  #  roles = ["member","load-balancer_member"]
  #  roles = [data.openstack_identity_role_v3.member.name, data.openstack_identity_role_v3.lb_member.name]
  # expires_at = timeadd(timestamp(), "10m") # valid for 10 minutes
}

output "demo_project_ac_secret" {
  value     = openstack_identity_application_credential_v3.ac_demo_project.secret
  sensitive = true
}
```

### Panic Output
```openstack_identity_application_credential_v3.ac_demo_project: Creating...
╷
│ Error: Plugin did not respond
│
│   with openstack_identity_application_credential_v3.ac_demo_project,
│   on users.tf line 23, in resource "openstack_identity_application_credential_v3" "ac_demo_project":
│   23: resource "openstack_identity_application_credential_v3" "ac_demo_project" {
│
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may contain more details.
╵

Stack trace from the terraform-provider-openstack_v1.43.0.exe plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x20 pc=0x105e09c]

goroutine 84 [running]:
github.com/terraform-provider-openstack/terraform-provider-openstack/openstack.getTokenInfoV3(0x144d020, 0xc0006a0e70, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x5b, 0x0)    
        github.com/terraform-provider-openstack/terraform-provider-openstack/openstack/identity_auth_scope_v3.go:170 +0x27c
github.com/terraform-provider-openstack/terraform-provider-openstack/openstack.getTokenInfo(0xc00093e5f0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
        github.com/terraform-provider-openstack/terraform-provider-openstack/openstack/identity_auth_scope_v3.go:147 +0xe9
github.com/terraform-provider-openstack/terraform-provider-openstack/openstack.resourceIdentityApplicationCredentialV3Create(0xc00094f180, 0x1479100, 0xc0007fe6c0, 0x2, 0x1e5b380)
        github.com/terraform-provider-openstack/terraform-provider-openstack/openstack/resource_openstack_identity_application_credential_v3.go:125 +0x1bb
github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*Resource).Apply(0xc00021aa00, 0xc00093f360, 0xc000948ec0, 0x1479100, 0xc0007fe6c0, 0x1309a01, 0xc000207d88, 0xc0006a09f0)
        github.com/hashicorp/terraform-plugin-sdk@v1.17.2/helper/schema/resource.go:320 +0x395
github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*Provider).Apply(0xc00017ee80, 0xc000957a38, 0xc00093f360, 0xc000948ec0, 0xc000691b88, 0xc000689c60, 0x130bb40)
        github.com/hashicorp/terraform-plugin-sdk@v1.17.2/helper/schema/provider.go:294 +0xa5
github.com/hashicorp/terraform-plugin-sdk/internal/helper/plugin.(*GRPCProviderServer).ApplyResourceChange(0xc0000074f0, 0x16e93f0, 0xc0006936b0, 0xc00094ebd0, 0xc0000074f0, 0xc0006936b0, 0xc000453ba0)
        github.com/hashicorp/terraform-plugin-sdk@v1.17.2/internal/helper/plugin/grpc_provider.go:895 +0x8c5
github.com/hashicorp/terraform-plugin-sdk/internal/tfplugin5._Provider_ApplyResourceChange_Handler(0x1454960, 0xc0000074f0, 0x16e93f0, 0xc0006936b0, 0xc0009469c0, 
0x0, 0x16e93f0, 0xc0006936b0, 0xc00069a000, 0x189)
        github.com/hashicorp/terraform-plugin-sdk@v1.17.2/internal/tfplugin5/tfplugin5.pb.go:3305 +0x222
google.golang.org/grpc.(*Server).processUnaryRPC(0xc00038c1c0, 0x16f4858, 0xc000584300, 0xc0006c2400, 0xc000283b90, 0x1dee360, 0x0, 0x0, 0x0)
        google.golang.org/grpc@v1.32.0/server.go:1194 +0x52b
google.golang.org/grpc.(*Server).handleStream(0xc00038c1c0, 0x16f4858, 0xc000584300, 0xc0006c2400, 0x0)
        google.golang.org/grpc@v1.32.0/server.go:1517 +0xd0c
google.golang.org/grpc.(*Server).serveStreams.func1.2(0xc00028ccb0, 0xc00038c1c0, 0x16f4858, 0xc000584300, 0xc0006c2400)
        google.golang.org/grpc@v1.32.0/server.go:859 +0xb2
created by google.golang.org/grpc.(*Server).serveStreams.func1
        google.golang.org/grpc@v1.32.0/server.go:857 +0x1fd

Error: The terraform-provider-openstack_v1.43.0.exe plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```
### Expected Behavior
Creation of Application Credential

### Actual Behavior
Openstack Plugin crashed

### Steps to Reproduce
1. `terraform apply`

--- a/openstack/dns_zone_v2.go
+++ b/openstack/dns_zone_v2.go
@@ -60,7 +60,7 @@ func dnsClientSetAuthHeader(resourceData *schema.ResourceData, dnsClient *gopher
 	// If project_id is different from auth one, set AuthSudo header
 	if v, ok := resourceData.GetOk("project_id"); ok {
 		if projectID, ok := v.(string); ok {
-			if project.ID != projectID {
+			if project != nil && project.ID != projectID {
 				headers[headerAuthSudoTenantID] = projectID
 			}
 		} else {

--- a/openstack/identity_auth_scope_v3.go
+++ b/openstack/identity_auth_scope_v3.go
@@ -167,7 +167,9 @@ func getTokenInfoV3(t interface{}) (authScopeTokenInfo, error) {
 			return info, err
 		}
 		info.userID = user.ID
-		info.projectID = project.ID
+		if project != nil {
+			info.projectID = project.ID
+		}
 		return info, nil
 	case tokens3.GetResult:
 		user, err := r.ExtractUser()
@@ -179,7 +181,9 @@ func getTokenInfoV3(t interface{}) (authScopeTokenInfo, error) {
 			return info, err
 		}
 		info.userID = user.ID
-		info.projectID = project.ID
+		if project != nil {
+			info.projectID = project.ID
+		}
 		return info, nil
 	default:
 		return info, fmt.Errorf("got unexpected AuthResult type %t", r)

